src/main/scala/cromwell/engine/WorkflowActor.scala

```diff
@@ -36,10 +34,13 @@ object WorkflowActor {
 case class WorkflowActor(id: WorkflowId,
                          namespace: WdlNamespace,
                          actualInputs: WorkflowCoercedInputs,
-                         backend: Backend) extends LoggingFSM[WorkflowState, WorkflowFailure] {
+                         backend: Backend) extends LoggingFSM[WorkflowState, WorkflowFailure] with WorkflowDescriptor {
+
   implicit val timeout = Timeout(5 seconds)
 
-  private val storeActor = context.actorOf(StoreActor.props(namespace, actualInputs))
+  private val storeActor = context.actorOf(StoreActor.props(namespace, backend.initializeWorkflow(this, actualInputs)))
```

Comment:
Right, I agree that it's an unlikely scenario although it does make me a bit uneasy simply because they bash you over the head to never, ever, ever, ever, ever pass `this` around and then we just hide behind a downcast to do so :)  OTOH it does seem overly cautious.

Thinking about the balance between the two a bit more, I suppose it does bother me a bit that the entire Actor (ignoring safety concerns) is being passed around (albeit behind a stripped down trait) just to get access to it's id and such.

What about a trait which our actor types could extend with a function that returns a case class w/ the values you're looking for? That seems like something which is in the wheelhouse of case class.